File: Segundo Periodo/AED2/TPs/arquivo.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

void tirarzeros(char *x) {
  int len = strlen(x);
  int naoNUlos = 0;
  int pos = 0;
  for (int i = 0; i < len; i++) {
    if (x[i] == '.') {
      pos = i;
    }
  }
  while (len > pos) {
    if (x[len - 1] != '0' && x[len - 1] != '.') {
      naoNUlos++;
    }
    if (naoNUlos == 0) {
      x[len - 1] = '\0';
    }
    if (x[len - 1] == '.' && naoNUlos == 0) {
      x[len - 1] = '\0';
      len = 0;
    }
    if (x[len - 1] == '.') {
      len = 0;
    }
    len--;
  }
}

void printar(char *x) {
  int len = strlen(x);
  if (x[len - 2] == '.' && x[len - 1] == '0') {
    len -= 2;
  }
  for (int i = 0; i < len; i++) {
    printf("%c", x[i]);
  }
  printf("\n");
}

int main() {
  int n;
  scanf("%d", &n);

  FILE *file = fopen("numbers.txt", "w");

  int i = 0;
  float nums;
  char texto[100];

  while (i < n) {
    scanf("%f", &nums);
    fprintf(file, "%.3f\n", nums);
    i++;
  }

  fclose(file);

  file = fopen("numbers.txt", "r");

  fseek(file, 0, SEEK_END);
  long len = ftell(file);
  fseek(file, 0, SEEK_SET);

  char letra;
  memset(texto, 0, sizeof(texto));
  long counter = 1;

  while (counter <= len) {
    fseek(file, -counter - 1, SEEK_END);
    letra = fgetc(file);

    if (letra == '\n' || counter == len) {
      tirarzeros(texto);
      printar(texto);
      memset(texto, 0, sizeof(texto));
    } else {
      if ((letra >= '0' && letra <= '9') || letra == '.') {
        memmove(texto + 1, texto, strlen(texto) + 1);
        texto[0] = letra;
      }
    }
    counter++;
  }

  fclose(file);

  return 0;
}

File: Segundo Periodo/AED2/TPs/arquivo.java
import java.io.RandomAccessFile;

public class arquivo {
    public static void printar(String x){
        int len = x.length();
        if(x.charAt(len-2) == '.' && x.charAt(len-1) == '0'){
            len -= 2;
        }
        for(int i = 0; i < len; i++){
            System.out.print(x.charAt(i));
        }
        System.out.println();
    }
    public static void main(String[] args) {
        int n = MyIO.readInt();
        int i = 0;
        float nums = 0;
        String texto = "";
        try {
            RandomAccessFile file = new RandomAccessFile("numbers.txt","rw");
            while(i < n){
                nums = MyIO.readFloat();
                file.writeChars(nums + "\n");
                i++;
            }
            file.close();
        } catch (Exception e) {
            e.printStackTrace();
        }
        try {
            RandomAccessFile file = new RandomAccessFile("numbers.txt","rw");
            long len = file.length();
            file.seek(file.length());
            char letra;
            texto = "";
            long counter = 1;
            while(counter < len){
                file.seek(len-counter-1);
                letra = (char)file.readByte();    
                if(letra == '\n'){
                    printar(texto);
                    texto = "";
                    counter++;
                }else{
                    texto = (('0' <= letra && letra <= '9') || letra == '.') ? (letra + texto) : texto;
                }
                counter++;
            }
            System.out.println(texto);
            file.close();
        } catch (Exception e) {
            System.out.println(e.getMessage());
        }
    }
}